common/rfb/CConnection.h

```diff
@@ -37,7 +37,13 @@ namespace rfb {
   class CSecurity;
   class IdentityVerifier;
 
-  class CConnection : public CMsgHandler {
+  class CConnection
+      : public CMsgHandler
+      , public UserPasswdGetter
+#if defined(HAVE_GNUTLS) || defined(HAVE_NETTLE)
+      ,public UserMsgBox
+#endif
```

Comment:
These interfaces are overkill now that the callbacks are part of an existing class. So please put the methods directly on `CConnection` and get rid of `USerPasswdGetter` and `UserMsgBox`.